.github/workflows/lint_and_test.yml

```diff
@@ -0,0 +1,33 @@
+name: Lint and test the project
+on:
+  pull_request:
+    types: [ opened, synchronize, edited, reopened, ready_for_review ]
+
+jobs:
+  lint_and_test:
+    name: Static code analyze & Unit test
+    runs-on: ubuntu-latest
+    environment: staging
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v2.3.2
+
+      - name: Set up Flutter with its version
+        uses: subosito/flutter-action@v1
+        with:
+          flutter-version: '3.3.8'
+
+      - name: Get flutter dependencies for the project
+        run: flutter pub get
+
+      - name: Run code generator script
+        run: flutter packages pub run build_runner build --delete-conflicting-outputs
+
+      - name: Check for any formatting issues in the code
+        run: flutter format --set-exit-if-changed .
+
+      - name: Statically analyze the Dart code.
+        run: flutter analyze
+
+      - name: Run widget tests, unit tests.
+        run: flutter test
```

Comment:
@nmint8m this step is for unittest only if I'm not wrong, so we might consider updating the ticket's description to get rid of `UI test` 😄